tcp/server.go

```diff
@@ -49,35 +57,36 @@ func getTCProot(root string) string {
 	return root
 }
 
-//log Error if one occures
-func checkErr(err error) {
-	if err != nil {
-		qbin.Log.Errorf("TCP Server Error: %s", err.Error())
-	}
-}
-
 //handle incomming TCP Connection
 func handleClient(connTCP net.Conn, root string) {
+	qbin.Log.Debug("Handeling incomming connection...")
 	//set timeout to 5 seconds, limit requestlength to 1MB, close connection on programm close
```

Comment:
I can upload files larger than 1MB. Maybe https://github.com/golang/go/blob/master/src/net/http/request.go#L730 is a starting point to add a MaxBytesReader that works with TCP (and maybe other stuff too, possibly a `MaxBytesReader(errorHandler func(), r io.ReadCloser, n int64)`?)